os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/wdt_api.c

```diff
@@ -100,17 +96,17 @@ void watchdog_init(uint32_t timeout_ms)
 	WDG_InitStruct.EICNT = timeout_ms > 100 ? 100 : timeout_ms / 2;
 	WDG_InitStruct.EIMOD     = ENABLE;
 
-	WDG_Init(WDGDev, &WDG_InitStruct);
+	WDG_Init(WDGArray[WDG_id], &WDG_InitStruct);
```

Comment:
I think we need to check WDG_id with bound of WDGArray to prevent out of memory.